### ISSUE TITLE
Add  error messages for require statements

### ISSUE THREAD

**Issue by paulinablaszk** [OpenZeppelin/ethernaut#147]
I think we should add error messages for all require (and assert) statements in contracts. What do you think about the format of reason strings? I think it should be consistent with other OpenZeppelin contracts

**Comment by abcoathup** [comment]
I agree that the format should be consistent with other OpenZeppelin contracts.

Should we wait until https://github.com/OpenZeppelin/ethernaut/issues/117 is done and then update or do it as part of that?  I am keen to get the upgrade to Solidity 0.5 complete.

**Comment by obernardovieira** [comment]
I don't have good experience with require error messages on ropsten. How about you guys?
And this seems to be the solution (or the problem) https://github.com/ethereum/go-ethereum/issues/18360
Also, we should wait until #117 is merged and then move on with everything else, in my opinion.

**Comment by paulinablaszk** [comment]
I agree we have to wait for the update to Solidity 0.5, I was thinking about adding error messages during #177 but I didn't want to mix these issues. 
I think that even if error messages are not supported by all clients, it might be a 
example of good practices for people learning Solidity, because it is helpful in the development process

**Comment by xaler5** [comment]
@paulinablaszk are there still requires without messages ?
